prober/derp_test.go

```diff
@@ -0,0 +1,102 @@
+package prober
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"tailscale.com/tailcfg"
+)
+
+func TestDerpProber(t *testing.T) {
+	dm := &tailcfg.DERPMap{
+		Regions: map[int]*tailcfg.DERPRegion{
+			0: {
+				RegionID:   0,
+				RegionCode: "zero",
+				Nodes: []*tailcfg.DERPNode{
+					{
+						Name:     "n1",
+						RegionID: 0,
+						HostName: "derpn1.tailscale.fake",
```

Comment:
The TLD ".test" is reserved for tests and guaranteed to never appear as a real TLD on the Internet.

https://en.wikipedia.org/wiki/.test